parlai/crowdsourcing/tasks/model_chat/analysis/compile_results.py

```diff
@@ -0,0 +1,522 @@
+        num_incomplete_convos = 0
+        num_complete_convos = 0
+        complete_convos_per_model = {}
+        bad_conversations = []
+        old_experimental_design_conversations = []
+        problem_counts = {}
+        worker_stats = {}
+        worker_conversation_counts = {}
+        total_utterances = 0
+
+        conversation_idx = 0
+        conversation_dfs = []
+        for read_folder in read_folders:
+            read_folder_name = os.path.split(read_folder)[-1]
+            for file_name in sorted(os.listdir(read_folder)):
+                if file_name in self.hit_block_list:
+                    continue
+
+                if 'incomplete' in file_name:
+                    num_incomplete_convos += 1
+                    continue
+                else:
+                    num_complete_convos += 1
+
+                # Read in file
+                with open(os.path.join(read_folder, file_name), 'rb') as f:
+                    data = json.load(f)
+
+                # Only include the first max_convos_per_worker conversations from a
+                # worker to avoid biasing
+                worker_id = data['workers'][0]
+                assignment_id = data['assignment_ids'][0]
+                if worker_id in worker_conversation_counts:
+                    conversations_so_far = worker_conversation_counts[worker_id]
+                else:
+                    conversations_so_far = 0
+                worker_conversation_counts[worker_id] = conversations_so_far + 1
+                if (
+                    self.max_convos_per_worker != -1
+                    and conversations_so_far >= self.max_convos_per_worker
+                ):
+                    print(
+                        f'Had {conversations_so_far} conversations already from this worker {worker_id}. Skipping {assignment_id}.'
+                    )
+                    continue
+
+                # Check if need to block the turker
+                word_counts = [
+                    len(d['text'].split(' '))
+                    for d in data['dialog']
+                    if d['agent_idx'] == 0
+                ]
+                utterances = [d['text'] for d in data['dialog'] if d['agent_idx'] == 0]
+                words = [
+                    d['text'].lower().split(' ')
+                    for d in data['dialog']
+                    if d['agent_idx'] == 0
+                ]
+                if np.average(word_counts) < self.min_word_count or (
+                    'contradiction' in self.PROBLEM_BUCKETS and 'contradiction' in words
+                ):
+                    bad_conversations.append(data)
+                    print(
+                        f'Bad complete conversation, words from human: {utterances}. Skipping.'
+                    )
+                    continue
+
+                if (
+                    'non_sensical' in self.PROBLEM_BUCKETS
+                    and 'non_sensical' not in data['dialog'][1]['problem_data']
```

Comment:
why is this old experimental design?

Reply:
Hmm, good point - I've made it more useful, by making it check that all of the desired buckets are present regardless of what the buckets are